repo/format/format_manager.go

```diff
@@ -0,0 +1,466 @@
+package format
+
+import (
+	"context"
+	"crypto/rand"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/internal/feature"
+	"github.com/kopia/kopia/internal/gather"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/encryption"
+	"github.com/kopia/kopia/repo/hashing"
+	"github.com/kopia/kopia/repo/logging"
+)
+
+var log = logging.Module("kopia/repo/format")
+
+// UniqueIDLength is the length of random unique ID of each repository.
+const UniqueIDLength = 32
```

Comment:
Need units.  Length in what?  Bytes, bits, words?

Reply:
done